File: Data Structures & Algorithms/longest-common-prefix/submission-2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        c: str = " "
        for col in range(0, len(strs[0])):
            c: str = strs[0][col]
            for row in range(0, len(strs)):
                if col >= len(strs[row]) or c != strs[row][col]:
                    return strs[0][:col]
        return strs[0]

File: Data Structures & Algorithms/valid-word-square/submission-2.cpp
class Solution {
public:
    bool validWordSquare(vector<string>& words) {
        size_t len = words[0].length();
        for(string word : words) {
            if(word.length() < len) {
                len = word.length();
            } else if(word.length() > len) {
                return false;
            }
        }
        string word;
        for(size_t row = 0; row < words.size(); ++row) {
            word = words[row];
            if(word.length() > words.size()) {
                return false;
            }
            for(size_t col = 0; col < word.length(); ++col) {
                if(word[col] != words[col][row]) {
                    return false;
                }
            }
        }
        return true;
    }
};
